dev-tools/xtask/src/illumos.rs

```diff
@@ -0,0 +1,144 @@
+use anyhow::{bail, Context, Result};
+use camino::Utf8Path;
+use cargo_metadata::Message;
+use fs_err as fs;
+use serde::Deserialize;
+use std::{
+    collections::{BTreeMap, BTreeSet, HashMap},
+    io::BufReader,
+    process::{Command, Stdio},
+};
+
+use crate::load_workspace;
+
+#[derive(Deserialize, Debug)]
+struct LibraryConfig {
+    binary_allow_list: Option<BTreeSet<String>>,
+}
+
+#[derive(Deserialize, Debug)]
+struct XtaskConfig {
+    libraries: BTreeMap<String, LibraryConfig>,
+}
+
+#[derive(Debug)]
+enum LibraryError {
+    Unexpected(String),
+    NotAllowed(String),
+}
+
+/// Verify that the binary at the provided path complies with the rules laid out
+/// in the xtask.toml config file. Errors are pushed to a hashmap so that we can
+/// display to a user the entire list of issues in one go.
+fn verify_executable(
+    config: &XtaskConfig,
+    path: &Utf8Path,
+    errors: &mut HashMap<String, Vec<LibraryError>>,
```

Comment:
Should be some kind of ordered map -- either a `BTreeMap` or an `IndexMap` (probably a `BTreeMap`)

Reply:
Fixed in [fa5b905](https://github.com/oxidecomputer/omicron/pull/5259/commits/fa5b905ef38a8a5f50130329967961514346ec67)